tsl/src/remote/row_by_row_fetcher.c

```diff
@@ -174,7 +174,13 @@ row_by_row_fetcher_complete(RowByRowFetcher *fetcher)
 
 			if (!(PQresultStatus(res) == PGRES_SINGLE_TUPLE ||
 				  PQresultStatus(res) == PGRES_TUPLES_OK))
+			{
+				/* remote_result_elog will call PQclear() on the result, so
```

Comment:
What about `pfree(AsyncResponseResult)`, is it not needed here? 

Maybe it would be simpler to remove this PQClear from remote_result_elog, if we wrap it in another try/catch anyway.

Reply:
AsyncResponseResult is manged by a PostgreSQL memory context, so isn't strictly needed, although we can  of course do that too.

The libpq library, which PQclear belongs to, is not using a memory context and is instead malloc'd so it is much more important to free that, although we have added extra functionality to also auto-free that memory on connection close if not freed earlier. Still, we'd like to PQclear results as soon as possible to not accumulate too much memory on long-lived connections.

I considered removing the PQclear from remote_result_elog but didn't want to do that now because I am unsure of the consequences for other places that use remote_result_elog.

I think the whole async library is in need of a refactoring, so would rather leave that for later and focus on fixing the immediate issue.

Reply:
added explicit pfrees